test/test_main.cpp

```diff
@@ -79,6 +77,31 @@ int main(int argc, char* argv[])
         return returnCode;
     }
 
+        // If we're talking about a particular OpenCL device,
+        // we have an API designed to tell us device capabilities and limitations,
+        // like maximum RAM allocation.
+#if ENABLE_OPENCL
+    if (opencl_single) {
+        // Make sure the context singleton is initialized.
+        CreateQuantumInterface(QINTERFACE_OPENCL, 1, 0).reset();
+
+        DeviceContextPtr device_context = OCLEngine::Instance()->GetDeviceContextPtr(device_id);
+        size_t maxMem = device_context->device.getInfo<CL_DEVICE_GLOBAL_MEM_SIZE>() / sizeof(complex);
+        size_t maxAlloc = device_context->device.getInfo<CL_DEVICE_MAX_MEM_ALLOC_SIZE>() / sizeof(complex);
+
+        // Device RAM should be large enough for 2 times the size of the stateVec, plus some excess.
+        max_qubits = log2(maxAlloc);
+        if ((3U * pow2(max_qubits)) > maxMem) {
```

Comment:
Magic number - can this be replaced with a constant, or even better a call into QInterface that returns some aspect of this?

Reply:
The '3U' is specifically a (constant) "property" of the design of QEngineOCL, hence I'm tempted to make it a static constant, there.

Reply:
Added it as a `static const`, in QEngineOCL.